addons/mail/models/mail_guest.py

```diff
@@ -29,6 +29,13 @@ def _lang_get(self):
     timezone = fields.Selection(string="Timezone", selection=_tz_get)
     channel_ids = fields.Many2many(string="Channels", comodel_name='mail.channel', relation='mail_channel_partner', column1='guest_id', column2='channel_id', copy=False)
 
+    @api.model
+    def validate_guest_name(self, name):
+        return (
+            type(name) is str and
+            0 < len(name) < 512
+        )
+
```

Comment:
Guideline is to make methods private unless necessary,
Have it single line,
I would expect a "validate" method to raise a ValidationError with an appropriate message, probably inside an actual `@api.constrains` instead of a manual check.